Furball.Engine/Engine/Helpers/MathHelper.cs

```diff
@@ -31,5 +32,41 @@ public static class MathHelper {
         /// <param name="deg">Amount in degrees</param>
         /// <returns>Amount in radians</returns>
         public static double DegreesToRadians(double deg) => deg * Math.PI / 180.0;
+        /// <summary>
+        /// Performs a Catmull-Rom interpolation using the specified positions.
+        /// </summary>
+        /// <param name="value1">The first position in the interpolation.</param>
+        /// <param name="value2">The second position in the interpolation.</param>
+        /// <param name="value3">The third position in the interpolation.</param>
+        /// <param name="value4">The fourth position in the interpolation.</param>
+        /// <param name="amount">Weighting factor.</param>
+        /// <returns>A position that is the result of the Catmull-Rom interpolation.</returns>
+        public static float CatmullRom(float value1, float value2, float value3, float value4, float amount)
+        {
+            // Using formula from http://www.mvps.org/directx/articles/catmull/
+            // Internally using doubles not to lose precission
+            double amountSquared = amount * amount;
+            double amountCubed = amountSquared * amount;
+            return (float)(0.5 * (2.0 * value2 +
+                                  (value3 - value1) * amount +
+                                  (2.0 * value1 - 5.0 * value2 + 4.0 * value3 - value4) * amountSquared +
+                                  (3.0 * value2 - value1 - 3.0 * value3 + value4) * amountCubed));
+        }
```

Comment:
This code is stolen from Monogame, and we 100% arent putting the right license here, while its MIT and while we should *probably* be fine, its better if we write our own

Reply:
im not sure how this is gonna change anything given its gonna just be formatted a little differently while working and looking exactly the same

Reply:
> im not sure how this is gonna change anything given its gonna just be formatted a little differently while working and looking exactly the same

What its going to change is the legality of it, as i said we'd probably be fine from what ive read but im not a legal expert and my opinion on whether its okay to use has changed from "its probably fine who cares" to "its probably file, but someone probably would care if it was wrong"

Reply:
Guess we have to rewrite the entirety of Color aswell then. Good luck!

Reply:
> Guess we have to rewrite the entirety of Color aswell then. Good luck!

Consider it done

Reply:
Color is fixed, now i just have to do Catmull Rom